programs/gateway/src/instructions/update_voter_weight_record.rs

```diff
@@ -41,6 +50,8 @@ pub fn update_voter_weight_record(
         &ctx.accounts.registrar.gatekeeper_network,
         None
     ).or(Err(error!(GatewayError::InvalidGatewayToken)))?;
+    
+    let voter_weight = extract_input_voter_weight(&ctx.accounts.input_voting_weight.to_account_info(), &ctx.accounts.registrar);
```

Comment:
When we have a previous plugin then  the data from its `VoteWeightRecord` (`voter_weight`, `weight_action`, `weight_action_target`, `reserved`) must be copied to our `VoteWeightRecord`

`voter_weight_expiry` should be set to `max`(`current_slot`, `previousVoteWeightRecord. voter_weight_expiry`) to either make the record valid only for the current slot or as long as the previous `VoteWeightRecord`
However the previous record expiry can't be longer then civic `gateway_token` validity if it has any expiration date